management/client.go

```diff
@@ -539,6 +539,68 @@ func (m *ClientManager) List(ctx context.Context, opts ...RequestOption) (c *Cli
 	return
 }
 
+// ListAll retrieves all the clients using offset pagination.
+// If limit is 0 it will retrieve all the clients, otherwise it will retrieve all the clients until the limit is reached.
+//
+// See: https://auth0.com/docs/api/management/v2#!/Clients/get_clients
+func (m *ClientManager) ListAll(ctx context.Context, limit int, opts ...RequestOption) (clients []*Client, err error) {
```

Comment:
Some further ideas for discussions over here on the topic of limit:

- A) Remove the limit entirely and always fetch everything
- B) Provide the limit as a designated RequestOption for ListAll funcs.

Reply:
Without a limit, this feature would not be very useful for the Auth0 CLI, because `list` commands accept a `-n` flag to limit the number of elements to output. Of course, we could still fetch it all and then drop the rest, but that's not particularly efficient nor DX-friendly (more requests than necessary take longer).

Reply:
A request option certainly seems like a good fit for it, instead of a regular argument.

Reply:
+1 on moving to a RequestOption rather than a dedicated part of the API

Reply:
I'm not sure getting the value from the context –for what is functionally an argument just to avoid defining it as such– would be a good fit. Do we have other similar examples in the SDK API?

I think getting the first X elements without having to handle the pagination is a legitimate enough use case. I wonder if it should be a separate method, though. Or if this method should be renamed to something that does not imply getting 'all', like `ListPaginated()`, and take the limit as a regular argument.

Reply:
After some chatting with @willvedd we figured that a `ListAll` that took an optional limit argument did feel a bit strange so keeping the function strictly as a "paginate all data" would fit better with the naming and avoid confusion

ce5a22a69eeccf26eb1e84e3a52ded73c8706f51 removes that

Reply:
@ewanharris and I had considered this. It's certainly a valid solution that would address all the technical concerns but I could also see it being too much being added at once. Frankly, as long as there is some `ListAll` for all lists, we're good; its not the worst thing if custom pagination exists in the Auth0 CLI codebase.

Reply:
I'd also be concerned, if we add a way to "get all" without also adding a way to "get N", that we could be encouraging "getting all" and then extracting the first N from the result (because it'll be easier than implementing the pagination from scratch).

Reply:
> Have you folks discussed this? @Widcket , @ewanharris , @willvedd ?

I'm afraid I wasn't part of those discussions.

Reply:
Fully agree with this @Widcket 👍🏻 , however we can split the 2 funcs in 2 separate PRs, but combine both in 1 release.